### DISCUSSION TITLE
CACHE_TIMEOUT removal and performance hit

### DISCUSSION THREAD

**Discussion by markkuleinio** [netbox-community/netbox#7457] [General]
In #6781 there is:

> Queryset caching is being removed in NetBox v3.0 (see #6639) due to frequent issues with cache invalidation. There's really no reason to keep it enabled by default for new deployments.

In my testing with 2.11.7 these are the results:

`CACHE_TIMEOUT = 0`:

```
netbox.ipam.prefixes.all():
Run 1: 37.89 seconds, for 10639 objects
Run 2: 37.37 seconds, for 10639 objects
Run 3: 37.19 seconds, for 10639 objects
```

`CACHE_TIMEOUT = 300`:

```
netbox.ipam.prefixes.all():
Run 1: 26.27 seconds, for 10639 objects
Run 2: 17.91 seconds, for 10639 objects
Run 3: 18.25 seconds, for 10639 objects
```

Again, this was in 2.11.7 and not the latest release.

Are there some other specific changes in NetBox 3.0 that improve performance even without this caching, or is it just so that the cache problems (?) are so common that the performance hit is just mandatory?

I totally get it that fetching all the objects at once is not ideal but in some cases it is just useful.

**Comment by jeremystretch** [comment]
There have been _repeated_ issues with cache invalidation ever since it was added, as discussed under #6639. (You can search the GitHub issues list for "caching" to get a sense for the breadth of the issue.)

**Reply by markkuleinio** [reply]
Right. I just saw some comments about getting better or same performance without caching, and wanted to say that's not something I see here. I actually disabled the caching first to fight the "SimpleLazyObject" problems, but it was the gunicorn's `threads=1` that fixed/prevented those I see.

**Reply by jeremystretch** [reply]
Bear in mind you're using pynetbox in your testing, which is going to add some additional overhead. Your pagination settings will affect performance as well.
